packages/flutter/lib/src/widgets/scroll_position.dart

```diff
@@ -307,7 +308,7 @@ class ScrollPosition extends ViewportOffset {
 
   @override
   void correctBy(double correction) {
-    _pixels += correction;
+    correctPixels(_pixels + correction);
```

Comment:
Oh I totally agree that this change is reasonable, I'm just reluctant to make the change before it's actually needed, since it is technically a perf regression (albeit a trivial one). It's more about the principle of the thing than actual performance concerns.